crates/cubecl-opt/src/lib.rs

```diff
@@ -107,13 +107,11 @@ pub struct ConstArray {
 #[derive(Default, Debug, Clone)]
 struct Program {
     pub const_arrays: Vec<ConstArray>,
-    pub variables: HashMap<(u16, u8), Item>,
-    pub(crate) slices: HashMap<(u16, u8), Slice>,
+    pub variables: HashMap<Id, Item>,
+    pub(crate) slices: HashMap<Id, Slice>,
```

Comment:
Do we need an hash map or we could use a vec instead? (really minor)

Reply:
I think it's better as a hashmap because it only tracks mutable variables/slices, so a vec would be extremely sparse. most variables will be immutable and not slices.